code/src/nuvla/job/job.py

```diff
@@ -41,6 +48,11 @@ def _init(self):
                 self.id = self.id.decode()
                 cimi_job = self.get_cimi_job(self.id)
                 dict.__init__(self, cimi_job)
+                if version_to_tuple(self.get('version')) > self._engine_version:
```

Comment:
> jobs with a version 2 should be able to run on a Job Engine 3.1.2?

Yes, as simple as that :)
I will change this to: for the job to be picked up it must be in the inclusive interval `[ major of job-eng-ver, job-eng-ver]`. Or we introduce a configuration that provides the range. Or we always ensure that the job engine can run jobs one version behind as well - this is to support situations when during the M->M+1 upgrade of job engine jobs with M can still be scheduled.
This will always skip jobs that don't have 'version' one them, because we agreed that the default, in this case, is '0' and falls outside of the meaningful interval.

Reply:
The comments are addressed in https://github.com/nuvla/job-engine/pull/149/commits/9805c082ac95a45f38389c5c6a24c3097fe97297